lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -165,11 +165,11 @@ def discover_cluster_uuid
     end
 
     def retrying_submit(actions)
-      # Initially we submit the full list of actions
-      submit_actions = actions
-
+      # filter out unsupported ES actions
+      submit_actions = filter_unsupported_actions(actions)
       sleep_interval = @retry_initial_interval
 
+      # Initially we submit the full list of valid actions
       while submit_actions && submit_actions.size > 0
```

Comment:
out-of-scope note: `submit_actions` is either an array that was given to this method, or the return value of `submit` which is also guaranteed to be an array (as long as our new `filter_unsupported_actions` does not return `nil`, which it can do as of cf9a834).

If we avoid returning `nil` from `filter_unsupported_actions`, this guard-clause (and several others in this method) can be simplified. _Feel free to exclude this suggestion from the scope of this PR, as it is merely meant to provide for context for other suggestions._

```suggestion
      while submit_actions.any?
```

Reply:
I agree with the parts for `unsupported actions` but in `retrying_submit` method using `.any?` will not be appropriative since `.any` doesn't work on `NilClass` (as my understanding). Here is my quick test:

```
[2022-08-29T11:33:56,621][ERROR][logstash.outputs.elasticsearch][main][ca10f590b...] Encountered an unexpected error submitting a bulk request, 
will retry {:message=>"undefined method `any?' for nil:NilClass", :exception=>NoMethodError, :backtrace=>["/Users/mashhur/Dev/elastic/logstash/logstash-plugins/logstash-output-elasticsearch/lib/logstash/plugin_mixins/elasticsearch/
common.rb:178:in `retrying_submit'", "/Users/mashhur/Dev/elastic/logstash/logstash-plugins/logstash-output-elasticsearch/lib/logstash/outputs/elasticsearch.rb:342:in `multi_receive'", "org/logstash/config/ir/compiler/AbstractOutputDelegatorExt.java:121:in `multi_receive'", "/Users/mashhur/Dev/elastic/logstash/logstash-core/lib/logstash/java_pipeline.rb:300:in `block in start_workers'"]}
```